pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/channel/ServiceUnitStateChannel.java

```diff
@@ -150,6 +151,18 @@ public interface ServiceUnitStateChannel extends Closeable {
      */
     CompletableFuture<Void> publishSplitEventAsync(Split split);
 
+    /**
+     * Asynchronously publishes the service unit unload event to the system topic in this channel,
+     * and wait to unload operation complete.
+     *
+     * @param unload (unload specification object)
+     * @param timeout The unload operation timeout.
+     * @param timeoutUnit The unload operation timeout unit.
+     * @return the completable future object staged from the event message sendAsync.
+     */
+    CompletableFuture<Void> publishUnloadEventAndWaitUnloadComplete(
```

Comment:
I don't think SUSC is responsible for waiting for the completion of events(too much coupling), so the waiting should be handled by `UnloadManager` outside of SUSC.

Can we pass the pub future to UnloadManager.handleUnload ? (I propose renaming `handleUnload` method to `waitAsync` too)
```
CompletedFuture<Void>  waitAsync(CompletedFuture<Void> unloadEventPubFuture, timeout, timeoutUnit){

return eventPubFuture.thenCompose(_ -> { 
inFlightUnloadRequest.computeIfAbsent(bundle, __ -> {
            if (log.isDebugEnabled()) {
                log.debug("Handle unload bundle: {}, timeout: {} {}", bundle, timeout, timeoutUnit);
 ...
        });

}
}


```

Reply:
Good point!